common/flatpak-dir.c

```diff
@@ -12703,20 +12703,60 @@ flatpak_dir_list_enumerated_remotes (FlatpakDir   *self,
   return (char **) g_ptr_array_free (g_steal_pointer (&res), FALSE);
 }
 
+typedef struct {
+  FlatpakDir *dir;
+  const char *prioritized_remote;
+} RemoteSortData;
+
+static gint
+cmp_remote_with_prioritized (gconstpointer a,
+                             gconstpointer b,
+                             gpointer      user_data)
+{
+  RemoteSortData *rsd = user_data;
+  FlatpakDir *self = rsd->dir;
+  const char *a_name = *(const char **) a;
+  const char *b_name = *(const char **) b;
+  int prio_a, prio_b;
+
+  prio_a = flatpak_dir_get_remote_prio (self, a_name);
+  prio_b = flatpak_dir_get_remote_prio (self, b_name);
+
+  /* Here we are assuming the array is already sorted by priority and only
+   * putting a particular remote at the top of its priority level */
+  if (prio_a == prio_b)
+    {
+      if (strcmp (a_name, rsd->prioritized_remote) == 0)
+        return -1;
+      if (strcmp (b_name, rsd->prioritized_remote) == 0)
+        return 1;
+    }
+
+  return 0;
```

Comment:
This isn't really a valid comparison function, as it doesn't define an order.

For example, if you had `a 0, b 1, c 1, d 2`.
Then a and d compare as equal, but they really need to have items between them.

So, even though we expect the list in prio order we need to compare the priorities anyway.

However for two items of the same prio we can return 0, because g_qsort is guaranteed to be stable (these days).

Reply:
Good point